ESP32main.ino

```diff
@@ -222,65 +236,61 @@ void enviar(void *parameter){
       switch (state_env){
       case 0:
               // No hacer nada de nada de res de res
-        break;  
+        break;
       case 1: { /*Enviar volumen*/
-        //Assignar volumen a la variable 
+        //Assignar volumen a la variable
         vTaskResume(esp32Client);
-        state_env = 2;
+        state_env = 2; //error
       }
-        break;      
-      case 2: { /*Esperar*/
-              
+        break;
+      case 2: { /*Esperar*/ //error
+
         //Serial.println("---->¿Quieres audio?");
         // Esperar respuesta
 
-        if(state == DATA){
+        if(state_env == WAITRESPONSE){
            state_env = 3;
```

Comment:
This isn't correct. For read or write in state_env variable we must use statesemaphore. Also, the state should be modified by client thread after sending the information. Finally, the state_env variable is modified just after being compared trough If statement which is redundant.